src/components/datatable.component.ts

```diff
@@ -452,6 +453,13 @@ export class DatatableComponent implements OnInit, AfterViewInit, DoCheck {
   @Output() rowContextmenu = new EventEmitter<{ event: MouseEvent, row: any }>(false);
 
   /**
+   * The context menu was invoked on a header cell.
+   *
+   * @memberOf DatatableComponent
+   */
+  @Output() headerContextmenu = new EventEmitter<{ event: MouseEvent, name: string, prop: string }>(false);
```

Comment:
That's very reasonable.

My thoughts on having a single event:
- Nicer API due to less outputs.
- Not backwards compatible: `rowContextmenu` should be renamed to `contextmenu`, the event interface will be changed.

If backwards compatibility is not a strong requirement, I'd prefer to implement your suggestion.